src/components/QueryEditor/QueryTypeEditor/index.tsx

```diff
@@ -30,13 +30,15 @@ export const QueryTypeEditor = ({ value }: Props) => {
   const dispatch = useDispatch();
 
   return (
-    <Segment
-      className={segmentStyles}
-      options={getQueryTypeOptions(datasource.getSupportedQueryTypes())}
-      onChange={(e) => {
-        isValidQueryType(e.value) && dispatch(changeQueryType(e.value!));
-      }}
-      value={toOption(value)}
-    />
+    <div data-testid="query-type-wrapper">
```

Comment:
Had to add the div wrapper as this is the only way to test Segment with selectEvent library. It doesn't change the UI at all

Reply:
seems like a bug in Segment! Feels like we should be able to pass it a testid, could be worth opening a ticket?